### PR TITLE
Add map rotation tests

### PR DESCRIPTION
**Pull request by strk**
Only checks raster for now, at +45 and -45 degrees.
Thanks Nyall for initial template.

### CODE REVIEW

**Comment by strk** [comment]
\cc @nyalldawson

**Comment by strk** [comment]
@nyalldawson I think the code here is ready but for some reason (I can't see right now) the tester returns non-zero. Sounds like a bug in the tester to fail w/out a reason.

See http://dash.orfeo-toolbox.org/testDetails.php?test=28515243&build=172691

File: tests/src/core/testqgsmaprotation.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+}
+
+void TestQgsMapRotation::linesLayer()
+{
+  mMapSettings.setLayers( QStringList() << mLinesLayer->id() );
+
+  // Arrowed line with parallel labels
+  QString qml = mTestDataDir + "lines_cardinals_arrowed_parallel_label.qml";
+  bool success = false;
+  mLinesLayer->loadNamedStyle( qml, success );
```
Comment (nyalldawson) [inline tests/src/core/testqgsmaprotation.cpp]: @strk it's this test which is failing - I think you've missed the associated qml file from the PR?

**Comment by strk** [comment]
Thanks, that was it. In master as 10d2ce45847d781719963ae8b49e21b6c84036db

**Pull request closed without merging**